### PR TITLE
Ron Test Cases (07) ALL

### PR DESCRIPTION
**Pull request by rontucuman**

### CODE REVIEW

File: src/main/java/org/umssdiplo/automationv01/core/managepage/addItem/AddItem.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        return itemTypeOptions;
+    }
+/*
+    public boolean isItemTypeInDropdown(String expectedItemType) {
+        List<WebElement> itemTypeOptions = getItemTypeOptions(itemTypeDropdown);
+        int index = getOptionIndex(itemTypeOptions, expectedItemType);
+
+        return index > -1;
+    }
+*/
```
Comment (lines17) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/addItem/AddItem.java:43]: Let's remove the comment

File: src/main/java/org/umssdiplo/automationv01/core/managepage/addItem/AddItem.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            statusDescr = "disabled";
+        }
+
+        return statusDescr;
+    }
+
+    private boolean getBtnStatus(WebElement btn) {
+        return btn.isEnabled();
+    }
+}
```
Comment (lines17) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/addItem/AddItem.java:85]: Let's do the Ctrl+Alt+L to remove extra spaces

File: src/main/java/org/umssdiplo/automationv01/core/managepage/navigationbar/ActivosMenu.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    public AddItemType clickAnadirTypeSubmenu() {
+        CommonEvents.clickButton(addItemTypeSubMenu);
+        return new AddItemType();
+    }
+
+    public AddItem clickAnadirItemSubmenu() {
+        CommonEvents.clickButton(addItemSubMenu);
+        return new AddItem();
+    }
+}
```
Comment (lines17) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/navigationbar/ActivosMenu.java:44]: I think it should contains their own package called  src/main/java/org/umssdiplo/automationv01/core/managepage/ActivosMenu/ActivosMenu.java

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSIApplication.java [hunk 4]
```diff
@@ -638,4 +648,141 @@ public void verifyThatTheGuardarButtonIsNotAvailableIfTheNombreDeAlmacenFieldIsE
...
+            login.enterPassword();
+        }
+    }
+
+    @And("^Click 'Login' button to accept credentials$")
+    public void clickLoginButtonToAcceptCredentials() throws Throwable {
+        if (navigationBar == null) {
+            navigationBar = login.clickLoginAcceptButton();
+        }
+    }
```
Comment (lines17) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSIApplication.java:678]: Let's reuse this steps  for login to the application : 
```
 @Given("^'SSI Application' page is loaded$")
    public void ssiApplicationPageIsLoaded() throws Throwable {
        login = LoadPage.loginPage();
    }

    @And("^set Admin credentials on 'Login' page$")
    public void setCredentialsOnLoginPage() throws Throwable {
        if (navigationBar == null) {
            navigationBar = login.setCredentials();
        }
    }
```

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSIApplication.java [hunk 5]
```diff
@@ -638,4 +648,141 @@ public void verifyThatTheGuardarButtonIsNotAvailableIfTheNombreDeAlmacenFieldIsE
...
+    @When("^Click 'Save' button$")
+    public void clickSaveButton() throws Throwable {
+        listItemType = editItemType.clickSaveItemTypeBtn();
+    }
+
+    @Then("^Verify row \"([^\"]*)\" has \"([^\"]*)\" as name in Item Type list page$")
+    public void verifyRowHasAsNameInItemTypeListPage(String indexParam, String expectedItemTypeName) throws Throwable {
+        int index = Integer.parseInt(indexParam) - 1;
+        String actualItemTypeName = listItemType.getItemTypeNameInTableRow(index);
+        Assert.assertEquals(actualItemTypeName, expectedItemTypeName);
```
Comment (lines17) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSIApplication.java:710]: Let's use the error messages for each assert. For example:
```
 @Then("^verify that new Contingency plan is added in the list$")
    public void verifyContingencyPLantWasAdded() throws Throwable {
        Assert.assertEquals(contingencyPlan.countContingencies(), totalContingencyPlans + NUMBER_ROW, String.format(ErrorMessage.ERROR_MESSAGE_CREATE_CONTINGENCY_PLAN, "contigency"));
```

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSIApplication.java [hunk 6]
```diff
@@ -638,4 +648,141 @@ public void verifyThatTheGuardarButtonIsNotAvailableIfTheNombreDeAlmacenFieldIsE
...
+
+    @When("^Click 'Delete' button of \"([^\"]*)\" row$")
+    public void clickDeleteButtonOfRow(String itemTypeName) throws Throwable {
+        listItemType = listItemType.clickDeleteBtnByItemTypeName(itemTypeName);
+    }
+
+    @Then("^Verify that \"([^\"]*)\" does not exist in the list$")
+    public void verifyThatDoesNotExistInTheList(String itemTypeName) throws Throwable {
+        boolean actualResult = listItemType.existItemTypeByName(itemTypeName);
+        Assert.assertEquals(actualResult, false);
```
Comment (lines17) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSIApplication.java:737]: Let's add error messages on your Asserts, please.

File: src/test/resources/features/ItemType.feature [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+  Scenario: RON003 TC - Ronald Tucuman - 03 - Verify that an Item Type List page can be  reached
+    Given 'SSI Application' page is loaded
+    #And Click 'Login' button on Page Header
+    #And Enter Username
+    #And Enter Password
+    #And Click 'Login' button to accept credentials
+    And Click 'Activos' menu in Navigation bar
+    When Click 'Lista de Tipos' submenu option
+    Then Verify that "Lista de Tipo de Item" is the title of the loaded page
```
Comment (lines17) [inline src/test/resources/features/ItemType.feature]: Let's remove the commented code

File: src/test/resources/features/ItemType.feature [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+    And Set "First Item Type" as the new name for the item type
+    When Click 'Save' button
+
+
+  Scenario: RON002 TC - Ronald Tucuman - 02 - Verify that an Item Type can be deleted
+    Given 'SSI Application' page is loaded
+    #And Click 'Login' button on Page Header
+    #And Enter Username
+    #And Enter Password
+    #And Click 'Login' button to accept credentials
```
Comment (lines17) [inline src/test/resources/features/ItemType.feature:26]: Let's use the steps definitions we are using on other scenarios:
```
Scenario: EO-14 Verificar que la pagina de la lista de posiciones es mostrada
    Given 'SSI' page is loaded
    And user is authenticated with administrator credentials
```

File: src/test/resources/features/ItemType.feature [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+    And Enter "Item Type for Testing 2" in Name input field
+    And Click 'Save' button
+    And Verify "Item Type for Testing 2" is in the list
+    When Click 'Delete' button of "Item Type for Testing 2" row
+    Then Verify that "Item Type for Testing 2" does not exist in the list
+
+
+  Scenario: RON003 TC - Ronald Tucuman - 03 - Verify that an Item Type List page can be  reached
+    Given 'SSI Application' page is loaded
+    #And Click 'Login' button on Page Header
```
Comment (CynthiaTerrazas) [inline src/test/resources/features/ItemType.feature:39]: remove comments

File: src/test/resources/features/ItemType.feature [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+    #And Enter Password
+    #And Click 'Login' button to accept credentials
+    And Click 'Activos' menu in Navigation bar
+    When Click 'Lista de Tipos' submenu option
+    Then Verify that "Lista de Tipo de Item" is the title of the loaded page
+
+
+  Scenario: RON004 TC - Ronald Tucuman - 04 - Verify that Cancel edit Item Item Type does not change Item Type name
+    Given 'SSI Application' page is loaded
+    #And Click 'Login' button on Page Header
```
Comment (CynthiaTerrazas) [inline src/test/resources/features/ItemType.feature:50]: remove comment for all scenarios

**Pull request closed without merging**